src/result.ts

```diff
@@ -17,18 +17,33 @@ export namespace Result {
    * arguments but returning `Ok` if successful, `Err` if the function throws
    *
    * @param fn function to wrap with ok on success or err on failure
-   * @param errorFn when an error is thrown, this will wrap the error result if provided
    */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  export function fromThrowable<Fn extends (...args: readonly any[]) => any>(
+    fn: Fn,
+  ): (...args: Parameters<Fn>) => Result<ReturnType<Fn>, unknown>
+  /**
+   * Wraps a function with a try catch, creating a new function with the same
+   * arguments but returning `Ok` if successful, `Err` if the function throws
+   *
+   * @param fn function to wrap with ok on success or err on failure
+   * @param errorFn when an error is thrown, this will wrap the error result
+   */
```

Comment:
It's not the most elegant to have to maintain two versions of the documentation, however I'm not aware of any other way of doing it. Please let me know if there's a better way!